app/Http/Controllers/UserController.php

```diff
@@ -125,4 +125,339 @@ public function destroy(User $user)
+        $user = factory(User::class)->create(
+            [
+                'name' => "Name Create",
+                'email' => 'email@gmail.com',
+                'password' => '12345678',
+                'role' => '1',
+                'avatar' => UploadedFile::fake()->image('avatar.jpg'),
+            ]
+        );
+        $params = [
+            'name' => 'name update',
+            'email' => null,
+            'password' => '12345678',
+            'role' => '1',
+            'avatar' => UploadedFile::fake()->image('avatar.jpg'),
+        ];
+        $response = $this->put(route('users.update', $user->id), $params);
+        $response
+            ->assertStatus(302)
+            ->assertSessionHasErrors('email');
+    }
+
+    public function testUpdateUserUniqueEmail()
+    {
+        $admin = factory(User::class)->create(['role' => UserType::ADMIN]);
+        $this->actingAs($admin);
+        $user1 = factory(User::class)->create(['email' => "emailduplicate@gmail.com"]);
+        $user = factory(User::class)->create(
+            [
+                'name' => "Name Create",
+                'email' => 'email@gmail.com',
+                'password' => '12345678',
+                'role' => '1',
+                'avatar' => UploadedFile::fake()->image('avatar.jpg'),
+            ]
+        );
+        $params = [
+            'name' => 'name update',
+            'email' => 'emailduplicate@gmail.com',
+            'password' => '12345678',
+            'role' => '1',
+            'avatar' => UploadedFile::fake()->image('avatar.jpg'),
+        ];
+        $response = $this->put(route('users.update', $user->id), $params);
+        $response
+            ->assertStatus(302)
+            ->assertSessionHasErrors('email');
+    }
+
+    public function testUpdateUserEmailMaximumLength()
+    {
+        $admin = factory(User::class)->create(['role' => UserType::ADMIN]);
+        $this->actingAs($admin);
+        $user = factory(User::class)->create(
+            [
+                'name' => "Name Create",
+                'email' => 'email@gmail.com',
+                'password' => '12345678',
+                'role' => '1',
+                'avatar' => UploadedFile::fake()->image('avatar.jpg'),
+            ]
+        );
+        $params = [
+            'name' => 'name update',
+            'email' => Str::random(201),
+            'password' => '12345678',
+            'role' => '1',
+            'avatar' => UploadedFile::fake()->image('avatar.jpg'),
+        ];
+        $response = $this->put(route('users.update', $user->id), $params);
+        $response
+            ->assertStatus(302)
+            ->assertSessionHasErrors('email');
+    }
+
+    public function testUpdateUserRequirePassword()
+    {
+        $admin = factory(User::class)->create(['role' => UserType::ADMIN]);
+        $this->actingAs($admin);
+        $user = factory(User::class)->create(
+            [
+                'name' => "Name Create",
```

Comment:
**Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 366</br>String "Name Create" does not require double quotes; use single quotes instead